openpype/lib/applications.py

```diff
@@ -736,13 +779,15 @@ class LaunchHook:
     # Order of prelaunch hook, will be executed as last if set to None.
     order = None
     # List of host implementations, skipped if empty.
-    hosts = []
-    # List of application groups
-    app_groups = []
-    # List of specific application names
-    app_names = []
-    # List of platform availability, skipped if empty.
-    platforms = []
+    hosts = set()
+    # Set of application groups
+    app_groups = set()
+    # Set of specific application names
+    app_names = set()
+    # Set of platform availability
+    platforms = set()
+    # Set of launch types for which is available
+    launch_types = set()
```

Comment:
What if we default this to `{LaunchTypes.local}` for  backwards compatibility, also for hooks outside of OpenPype codebase currently. It'd also require much less changes for this PR across much less files?